src/MagicWormhole/Internal/FileTransfer.hs

```diff
@@ -20,19 +20,28 @@ import Data.Aeson
   , object
   , withObject
   )
+import System.Posix.Types (FileOffset)
 
 -- | An offer made by a sender as part of the Magic Wormhole file transfer protocol.
 --
 -- Currently only supports sending simple text messages. A full version would
 -- also support sending files and directories.
-newtype Offer
+data Offer
```

Comment:
Hm!  I'm still learning about `newtype` and trying to understand when it's a good idea to use `newtype` and when it's a good idea to use `data`.  Could you help me understand the change in this case?

Reply:
`newtype` is like type aliases but with more type safety and zero cost (of wrapping/unwrapping that you might expect) provided the type has only one constructor with one field. In the original code, there was only possible constructor with just one field (i.e. `Message`), so `data` and `newtype` could be interchanged. However, we added the `File` constructor with two fields inside below, so we can no longer use `newtype`.